pkg/api/componentreadiness/component_report.go

```diff
@@ -415,45 +453,49 @@ func (c *componentReportGenerator) getCommonTestStatusQuery(allJobVariants crtyp
 		queryString += ` AND NOT 'Disruption' in UNNEST(capabilities)`
 	}
 
-	variantGroups := c.IncludeVariants
-	// potentially cross-compare variants for the sample
-	if isSample && len(c.VariantCrossCompare) > 0 {
-		variantGroups = c.CompareVariants
-	}
-	if variantGroups == nil { // server-side view definitions may omit a variants map
-		variantGroups = map[string][]string{}
-	}
+	// fallback queries get all variants with no filtering
+	// so all tests are fetched then cached
+	if !isFallback {
```

Comment:
Is there a in-code replacement for the removed IncludeVariants query filtering? Presumably if we skip it here we must do it in code later, but I couldn't find it in the PR. (searching for uses of IncludeVariants)

And followup question, can this function just change to always NOT filter on variants/capability here, all the time, regardless if it's fallback. Wondering if we can avoid a logic fork and have just one path through here, and always rely on our own code for the filtering.

Reply:
Capturing the discussion we had on this, swapping out the basis for the sample in the loop leaves us unable to detect missing samples.  Will need some work there to report on tests that existed in the base but not the sample when we look to refactor this.